src/control_interface.cpp

```diff
@@ -298,6 +306,92 @@ ControlInterface::ControlInterface(rclcpp::NodeOptions options) : Node("control_
 }
 //}
 
+/* parametersCallback //{ */
+rcl_interfaces::msg::SetParametersResult ControlInterface::parametersCallback(const std::vector<rclcpp::Parameter> &parameters){
+  rcl_interfaces::msg::SetParametersResult result;
+  result.successful = false;
+  result.reason = "";
+  char buff[300];
+
+  for (const auto &param: parameters){
+
+    /* takeoff_height //{ */
+    if (param.get_name() == "takeoff_height"){
+      if (param.get_type() == rclcpp::ParameterType::PARAMETER_DOUBLE){
+        if (param.as_double() >= 0.5 && param.as_double() < 10){
+          takeoff_height_ = param.as_double();
+          result.successful = true;
+          RCLCPP_INFO(this->get_logger(), "[%s]: Parameter: '%s' set to %1.2f", this->get_name(), param.get_name().c_str(), param.as_double());
+        }else{
+          snprintf(buff, sizeof(buff), "parameter '%s' cannot be set to %1.2f because it is not in range <0.5;10>", param.get_name().c_str(), param.as_double());
+          result.reason = buff;
+        }
+      }else{
+        snprintf(buff, sizeof(buff), "parameter '%s' has to be type DOUBLE", param.get_name().c_str());
+        result.reason = buff;
+      }
+    //}
+
+    /* waypoint_marker_scale //{ */
+    }else if (param.get_name() == "waypoint_marker_scale"){
+      if (param.get_type() == rclcpp::ParameterType::PARAMETER_DOUBLE){
+        if (param.as_double() > 0.0){
+          waypoint_marker_scale_ = param.as_double();
+          result.successful = true;
+          RCLCPP_INFO(this->get_logger(), "[%s]: Parameter: '%s' set to %1.2f", this->get_name(), param.get_name().c_str(), param.as_double());
+        }else{
+          snprintf(buff, sizeof(buff), "parameter '%s' cannot be set to %1.2f because it is not >0", param.get_name().c_str(), param.as_double());
+          result.reason = buff;
+        }
+      }else{
+        snprintf(buff, sizeof(buff), "parameter '%s' has to be type DOUBLE", param.get_name().c_str());
+        result.reason = buff;
+      }
+    //}
+
+    /* waypoint_loiter_time //{ */
+    }else if (param.get_name() == "waypoint_loiter_time"){
+      if (param.get_type() == rclcpp::ParameterType::PARAMETER_DOUBLE){
+        if (param.as_double() >= 0.0){
+          waypoint_loiter_time_ = param.as_double();
+          result.successful = true;
+          RCLCPP_INFO(this->get_logger(), "[%s]: Parameter: '%s' set to %1.2f", this->get_name(), param.get_name().c_str(), param.as_double());
+        }else{
+          snprintf(buff, sizeof(buff), "parameter '%s' cannot be set to %1.2f because it is not positive value", param.get_name().c_str(), param.as_double());
```

Comment:
I would change the wording a bit, to make it clear that loiter time can be zero.